src/odemis/acq/stream/_sync.py

```diff
@@ -1303,12 +1309,17 @@ def _runAcquisitionEbeam(self, future):
                             self._sccd._onNewData(self._ccd_df, self._acq_data[self._ccd_idx][-1])
                         except Exception:
                             logging.exception("Failed to update CCD live view")
+                        # check if we need to integrate images (integration count > 1)
+                        if integration_count > 1:
+                            # integrate the acquired images one after another during the acquisition
+                            self.integration_step += 1
+                            for stream_idx, das in enumerate(self._acq_data):
+                                self._acq_data[stream_idx], self.best_dtype = \
+                                    img.integrateImages(das, integration_count, self.integration_step,
+                                                        best_dtype=self.best_dtype)
```

Comment:
You cannot have a single best_dtype for all the streams, it's different for each of them (in theory). Same thing for the image integrator. I think you need to init with `self._img_intor = [None for s in streams]`, and in this loop here, create an image integrator if there is none yet. After the acquisition, it must be reset to `None` (both to save memory, and to make sure it still works the same on the next acquisition).